src/fighter_state.rs

```diff
@@ -1725,18 +1772,31 @@ fn grabbing(
                                     Some(items_assets.get(item).expect("Item not loaded!").clone());
                                 commands.entity(item_ent).despawn_recursive();
                             }
-                            ItemKind::BreakableBox { .. } => {
+                            ItemKind::BreakableBox { .. } | ItemKind::Bomb { .. } => {
                                 // Transition to holding state
                                 transition_intents.push_back(StateTransition::new(
-                                    Holding { item: item.clone() },
+                                    Holding,
                                     Holding::PRIORITY,
                                     true,
                                 ));
 
+                                let image = items_assets
+                                    .get(item)
+                                    .expect("Item not loaded!")
+                                    .clone()
+                                    .image;
+
+                                commands.entity(item_ent).insert(Transform::from_xyz(
+                                    0.,
+                                    consts::THROW_ITEM_OFFSET.y + image.image_size.y,
+                                    consts::PROJECTILE_Z,
+                                ));
+
                                 picked_item_ids.insert(item_ent);
                                 **fighter_inventory =
                                     Some(items_assets.get(item).expect("Item not loaded!").clone());
-                                commands.entity(item_ent).despawn_recursive();
+                                commands.entity(item_ent).remove::<Item>().insert(BeingHeld);
+                                commands.entity(fighter_ent).add_child(item_ent);
```

Comment:
I like the new way of doing Holding by adding as child. :+1: 
Nice job simplifying.